### ISSUE TITLE
I want to set Refill, Bandwidth capacity to 0.

### ISSUE THREAD

**Issue by BaeJi77** [vladimir-bukhtoyarov/bucket4j#135]
I took below error stack. 
``` java
java.lang.IllegalArgumentException: 0 is wrong value for period tokens, because tokens should be positive

at io.github.bucket4j.BucketExceptions.nonPositivePeriodTokens(BucketExceptions.java:106)
at io.github.bucket4j.Refill.<init>(Refill.java:46)
at io.github.bucket4j.Refill.greedy(Refill.java:95)
```

---

I read code related this issue. And find when capacity is set to 0, this library make exception.
``` java
public static Bandwidth classic(long capacity, Refill refill) {
        if (capacity <= 0) {
            throw BucketExceptions.nonPositiveCapacity(capacity);
        }
        if (refill == null) {
            throw BucketExceptions.nullBandwidthRefill();
        }
        return new Bandwidth(capacity, refill.periodNanos, refill.tokens, capacity, refill.refillIntervally, refill.timeOfFirstRefillMillis, refill.useAdaptiveInitialTokens);
    }
```

``` java
private Refill(long tokens, Duration period, boolean refillIntervally, long timeOfFirstRefillMillis, boolean useAdaptiveInitialTokens) {
        if (period == null) {
            throw BucketExceptions.nullRefillPeriod();
        }
        if (tokens <= 0) {
            throw BucketExceptions.nonPositivePeriodTokens(tokens);
        }
        this.periodNanos = period.toNanos();
        if (periodNanos <= 0) {
            throw BucketExceptions.nonPositivePeriod(periodNanos);
        }
...
}
```

Now I can set dynamically bucket size set. I sometimes want to stop to generate token. 

How can I do this situation? Do you have any ideas?

**Comment by vladimir-bukhtoyarov** [comment]
Hello @BaeJi77 

You can set a very slow refill for such case like following:
```java
Refill.gready(1, Duration.ofNanos(Long.MAX_VALUE));
```

**Comment by BaeJi77** [comment]
Thank you for answer!! :) 
I have other question!!

Can I reduce token when I change capacity?

ex) situation
1. Create new bucket that is capacity 100.
2. Reconfig to reduce bucket capacity to 1.
3. Continuesly total number of token is 100 in bucket. 

I used this version `compile 'com.github.vladimir-bukhtoyarov:bucket4j-core:4.10.0'`
Is it normal? I used `BucketConfiguration` object and `replaceConfiguration` method.

``` java
Bandwidth bandwidth = Bandwidth.classic(10, Refill.greedy(10, Duration.ofMinutes(1)));
        LocalBucket bucket = Bucket4j.builder()
                                    .addLimit(bandwidth)
                                    .build();

        System.out.println(bucket.getAvailableTokens());
        System.out.println(bucket.getConfiguration());

        Bandwidth newBandwidth = Bandwidth.classic(1, Refill.greedy(1, Duration.ofMinutes(1)));
        BucketConfiguration bucketConfiguration = Bucket4j.configurationBuilder()
                                                          .addLimit(newBandwidth)
                                                          .build();

        bucket.replaceConfiguration(bucketConfiguration);

        System.out.println(bucket.getAvailableTokens());
        System.out.println(bucket.getConfiguration());
```

``` java
10
BucketConfiguration{bandwidths=[Bandwidth{capacity=10, initialTokens=10, refillPeriodNanos=60000000000, refillTokens=10, refillIntervally=false, timeOfFirstRefillMillis=-9223372036854775808, useAdaptiveInitialTokens=false}]}

10
BucketConfiguration{bandwidths=[Bandwidth{capacity=1, initialTokens=1, refillPeriodNanos=60000000000, refillTokens=1, refillIntervally=false, timeOfFirstRefillMillis=-9223372036854775808, useAdaptiveInitialTokens=false}]}
```

I expect `bucket.getAvailableTokens()` this result return `1`. 
How can I resolve this situation?

**Comment by vladimir-bukhtoyarov** [comment]
It is a bug. I will provide a fix tomorrow

**Comment by BaeJi77** [comment]
I didn't understand what you were saying.
Is it not necessary to change the internal logic for the problem?

**Comment by vladimir-bukhtoyarov** [comment]
Desired behaviour is already implemented, but it does not work because of bug.

**Comment by BaeJi77** [comment]
Thanks for answering so many questions!

There seems to be a typo in document. How can I contribute to this?

**Comment by vladimir-bukhtoyarov** [comment]
According to this issue, I am going to fix it by myself, because it needs to be combined with another enhancement that was described in this thread https://github.com/vladimir-bukhtoyarov/bucket4j/issues/133

If you want to fix a documentation just fork and provide pull-request.

**Comment by BaeJi77** [comment]
Thank you! See you next time!

**Comment by vladimir-bukhtoyarov** [comment]
Has been released under the version ```6.0.0```.  Feel free to as any questions according to new API.
